Add guides to enable virtualization for Windows (#16)
upleveled/system-setup: windows.md

    It should print out a welcome message like this:<br><br>
    <img src="macos-6-docker-hello-world.png"><br><br>

    If after running `docker run hello-world` command, you receive an error like this:

    ```bash
    Unable to write to the database. Exit code: 1
      at Docker.Backend.Database.WriteKeys(ICollection\'1 keyValues) at Docker.Backend.Database.ModifyDatabase(ICollection`1 keyValuesToWrite, ICollection`1 keysToRemove)
      at Docker.Backend.Database.Write(Settings settings)
      at Docker.Backend.ContainerEngine.Linux.DoStart(Settings settings)
      at Docker.Backend.ContainerEngine.Linux.Start(Settings settings)
      at Docker.Core.Pipe.NamedPipeServer.<>c**DisplayClass8_0.<Register>b**0(Object[] parameters)
      at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters)
    ```

>     1. Press <kbd>Ctrl</kbd>-<kbd>Alt</kbd>-<kbd>Delete</kbd> and click on Task Manager
> 
>     2. Click on the Performance tab and make sure your CPU has the virtualisation setting enabled
>        <img src="windows-7-virtualisation-enabled.png"><br><br>
> 
>     3. In the case virtualization is not enabled, make sure your machine meets the [docker requirements](https://docs.docker.com/desktop/windows/troubleshoot/#virtualization) and follow the steps in the Microsoft Windows [official documentation page](https://support.microsoft.com/en-us/windows/enable-virtualization-on-windows-11-pcs-c5578302-6e43-4b4b-a449-8ced115f58e1)

Reviewer: ```suggestion
    Check if virtualization is enabled on your machine on the **Performance** tab of the Task Manager (<kbd>Ctrl</kbd>-<kbd>Shift</kbd>-<kbd>Escape</kbd> to open the Task Manager

    <img src="windows-7-virtualisation-enabled.png">

    If virtualization is not enabled, [enable VT-x or SVM in your UEFI or BIOS](https://example.com/add-link-here).
```

    It should print out a welcome message like this:<br><br>
    <img src="macos-6-docker-hello-world.png"><br><br>

>     If after running `docker run hello-world` command, you receive an error like this:

Reviewer: ```suggestion
    If running `docker run hello-world` command results in [an `Unable to write to the database` error](https://github.com/docker/for-win/issues/74#issuecomment-247665026) like the one below, it may mean that virtualization is not enabled on your machine.
```

    ```bash
    Unable to write to the database. Exit code: 1
      at Docker.Backend.Database.WriteKeys(ICollection\'1 keyValues) at Docker.Backend.Database.ModifyDatabase(ICollection`1 keyValuesToWrite, ICollection`1 keysToRemove)
      at Docker.Backend.Database.Write(Settings settings)
      at Docker.Backend.ContainerEngine.Linux.DoStart(Settings settings)
      at Docker.Backend.ContainerEngine.Linux.Start(Settings settings)
      at Docker.Core.Pipe.NamedPipeServer.<>c**DisplayClass8_0.<Register>b**0(Object[] parameters)
      at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters)
    ```

    1. Press <kbd>Ctrl</kbd>-<kbd>Alt</kbd>-<kbd>Delete</kbd> and click on Task Manager

>        <img src="windows-7-task-manager.webp"><br><br>

Reviewer: this image is placed in an unusual place and doesn't really add anything to the instructions

![Screen Shot 2022-04-30 at 12 57 50](https://user-images.githubusercontent.com/1935696/166102692-d303119e-ce6b-449d-a311-35547d422115.png)

Reviewer: removed in 19ef4d8

    2. Click on the Performance tab and make sure your CPU has the virtualisation setting enabled
       <img src="windows-8-virtualisation-enabled.png"><br><br>

>     3. In the case virtualization is not enabled, make sure your machine meets the [docker requirements](https://docs.docker.com/desktop/windows/troubleshoot/#virtualization) and follow the steps in the Microsoft Windows [official documentation page](https://support.microsoft.com/en-us/windows/enable-virtualization-on-windows-11-pcs-c5578302-6e43-4b4b-a449-8ced115f58e1)

Reviewer: capitaliztion of `Docker`

Author: https://github.com/upleveled/system-setup/pull/16/commits/8157cbb674694df4b5a2d0a9512cbdb6af4c564c

    <img src="windows-7-virtualisation-enabled.png"><br><br>

>     Only if virtualization is not enabled, **Enable VT-x or SVM in your UEFI or BIOS**

Reviewer: we can leave this as it was before (remove "Only" and remove the bold, add a colon at the end of the line)

Reviewer 2: e2aff03cb9f0c7ed1b735fd0c8eabb0e11148841

>     1. Reboot your computer

Reviewer: - [ ] let's use `Restart` instead of `Reboot`, simpler language that more people will be familiar with
- [ ] you should probably explain the part about the black screen and the keys to press earlier before telling people to restart their computer (because they may wait for Windows to completely load before reading further)

Reviewer 2: 15293d790179e6b6e588e046092db9308d2b4e74

>     2. Right when the computer is coming up from the black screen, press <kbd>Delete</kbd>, <kbd>Esc</kbd>, <kbd>F1</kbd>, <kbd>F2</kbd>, or <kbd>F4</kbd>. Each computer manufacturer uses a different key but it may show a brief message at boot telling you which one to press. If you miss it the first time, reboot and try again. It helps to tap the key about twice a second when the computer is coming up. If you are not able to enter the BIOS via this method, consult your computer’s manual.

Reviewer 2: Remove qoute
bb644e56e4d3ed1b7f2644ff407197f12eada677

    3. In the BIOS settings, find the configuration items related to the CPU. These can be in under the headings Processor, Chipset, or Northbridge.
    4. Enable virtualization; the setting may be called VT-x, AMD-V, SVM, or Vanderpool. Enable Intel VT-d or AMD IOMMU if the options are available.

>     5. Save your changes and reboot.

Reviewer 2: Remove periods
https://github.com/upleveled/system-setup/pull/16/commits/eebc418b9380172b9fd5744d62995216940cc4b9

>     If you are unable to find the Virtualization settings in your BIOS [make sure that your laptop does support it](https://archive.ph/aZ5Ng).

Reviewer: - [ ] No need to capitalize "Virtualization" here
- [ ] It's not the "laptop" that supports virtualization
- [ ] This link seems like a very long explanation to check whether the CPU supports

Reviewer 2: 3ce28a1e9d4354dbf06f43099176e836b7b78f1f
1b8aecccbaffcdd6c5e67dc671ef816bd678dbb7
9c0284c2dd351e856c25b6e4b890be7323117320

    When opening Docker Desktop during Option A or Option B, an error message `Docker Desktop - Windows Hypervisor is not present` may appear:<br><br>

    <img src="windows-7-docker-desktop-error.avif"><br><br>

    If this appears for you, follow the next steps to enable virtualization on your machine (if you don't receive the error, you can skip to the Docker testing step).

    1. Close PowerShell and open it again as administrator (like in step 2). Copy the following text and right-click in the blue middle part of the PowerShell window to paste the text. Hit enter.

       ```bash
       Get-ComputerInfo -Property HyperVRequirementVMMonitorModeExtensions,HyperVRequirementVirtualizationFirmwareEnabled | Format-List
       ```

       This should display the following output:

       ```bash
       HyperVRequirementVMMonitorModeExtensions: True
       HyperVRequirementVirtualizationFirmwareEnabled: False
       ```

       Ensure that your output is the same as above before proceeding with the steps below to enable virtualization

    2. Restart your machine

>     3. As soon as the monitor turns black during restart, press the BIOS key or UEFI key for your machine repeatedly. If you're not sure what that key is, either try to read the key on the screen quickly as your machine restarts or refer to [hotkey documentation online](https://kb.wisc.edu/helpdesk/page.php?id=58779) (common keys are <kbd>Delete</kbd>, <kbd>Esc</kbd>, <kbd>F1</kbd>, <kbd>F2</kbd>, <kbd>F9</kbd>, <kbd>F10</kbd> or <kbd>F12</kbd>)

Reviewer: ```suggestion
    3. As soon as the monitor turns black during restart, press the BIOS key or UEFI key for your machine repeatedly. If you're not sure what that key is, either try to read the key on the screen quickly as your machine restarts or refer to BIOS key documentation online eg. [the guide by the University of Wisconson-Madison](https://kb.wisc.edu/helpdesk/page.php?id=58779) (common keys are <kbd>Delete</kbd>, <kbd>Esc</kbd>, <kbd>F1</kbd>, <kbd>F2</kbd>, <kbd>F9</kbd>, <kbd>F10</kbd> or <kbd>F12</kbd>)
```

>     4. Optional: you may need to find and select an option to enter the `BIOS / UEFI Setup Utility`

Reviewer: ```suggestion
    4. On some computers, you will need to find and select the option to enter the BIOS / UEFI after hitting the hotkey
```

>     4. On some computers, you will need to find and select the option to enter the BIOS / UEFI after hitting the hotkey

Reviewer: ```suggestion
    4. On some machines, you will need to find and select the option to enter the BIOS / UEFI after hitting the hotkey
```

    4. Open the start menu and search for "Ubuntu". Start it - it should ask you to create a user with a password. This will be your user to log in to your Ubuntu Linux Subsystem - note down the username and password somewhere secure to make sure you do not forget it.
    5. Open the start menu and search for "Docker Desktop". Start it and go to the Settings. Under the General tab, you will find an option called "Use WSL 2 based engine". Make sure this is checked.

>     If opening Docker Desktop from **Option A or B** shows an error message that says `Docker Desktop - Windows Hypervisor is not present`, this indicates that virtualization needs to be enabled on your machine
> 
>     Only continue with the rest of these steps if you see the above error message. If you do not see this error message, you can continue with step `21`

Reviewer: ```suggestion
    When opening Docker Desktop during Option A or Option B, an error message `Docker Desktop - Windows Hypervisor is not present` may appear:<br><br>
    
    <img src="image here"><br><br>

    If this appears for you, follow the next steps to enable virtualization on your machine (if you don't receive the error, continue with the step to test Docker below):
```

    4. Open the start menu and search for "Ubuntu". Start it - it should ask you to create a user with a password. This will be your user to log in to your Ubuntu Linux Subsystem - note down the username and password somewhere secure to make sure you do not forget it.
    5. Open the start menu and search for "Docker Desktop". Start it and go to the Settings. Under the General tab, you will find an option called "Use WSL 2 based engine". Make sure this is checked.

    When opening Docker Desktop during Option A or Option B, an error message `Docker Desktop - Windows Hypervisor is not present` may appear:<br><br>
    
    <img src="image here"><br><br>

>     If this appears for you, follow the next steps to enable virtualization on your machine (if you don't receive the error, continue with the step below to test Docker):

Reviewer: ```suggestion
    If this appears for you, follow the next steps to enable virtualization on your machine (if you don't receive the error, continue with the step further below to test Docker):
```

>     If this appears for you, follow the next steps to enable virtualization on your machine (if you don't receive the error, continue with the step further below to test Docker):

Reviewer: ```suggestion
    If this appears for you, follow the next steps to enable virtualization on your machine (if you don't receive the error, you can skip to the Docker testing step):
```

>     Check if your machine supports virtualization by running the following command on powershell as an administrator:

Reviewer: check for casing of project names

(in the full changes of the whole PR)

Reviewer 2: I had only the PowerShell written wrong
566bd9e0543d9f18bfe43cfb34bf540be299aa90

    ```bash
    Get-ComputerInfo -Property HyperV*
    ```

    This should display the following output:

    ```bash
    HyperVisorPresent: False
    HyperVRequirementDataExecutionPreventionAvailable : True
    HyperVRequirementSecondLevelAddressTranslation    : True
    HyperVRequirementVirtualizationFirmwareEnabled    : False
    HyperVRequirementVMMonitorModeExtensions          : True
    ```

    `HyperVRequirementVMMonitorModeExtensions: True` means that your machine supports virtualization and `HyperVRequirementVirtualizationFirmwareEnabled: False` means that virtualization is not enabled

>     If your output is different than the above, please contact the UpLeveled team on Slack for further assistance

Reviewer 2: 2465d51de575c0d7a4b0bbd1b307117bfab332cb

>     If your output is different than the above, please contact the UpLeveled team on Slack for further assistance

Reviewer: new version is this:

```
Ensure that your output is the same as above before proceeding with the steps below to enable virtualization
```

we lost these two explanations:

means that your machine supports virtualization and virtualization is not enabled

>     If your output is different than the above, please contact the UpLeveled team on Slack for further assistance

Reviewer 2: 6110e52ee676373be853b055d98a45de013f1be7

>     Enable virtualization in your BIOS / UEFI settings by following these steps:

Reviewer: once you [make the "Check if your machine supports virtualization" the new step 1 above](https://github.com/upleveled/system-setup/pull/16/files/ca0070e6b6f4bd4d06ccdd96c6f0f32cff8a15da#r1561132600), then you can probably remove this paragraph

    1. Restart your machine
    2. As soon as the monitor turns black during restart, press the BIOS key or UEFI key for your machine repeatedly. If you're not sure what that key is, either try to read the key on the screen quickly as your machine restarts or refer to [hotkey documentation online](https://kb.wisc.edu/helpdesk/page.php?id=58779) (common keys are <kbd>Delete</kbd>, <kbd>Esc</kbd>, <kbd>F1</kbd>, <kbd>F2</kbd>, <kbd>F9</kbd>, <kbd>F10</kbd> or <kbd>F12</kbd>)
    3. Navigate to the `BIOS / UEFI Setup Utility`
    4. Find the virtualization option for your machine and enable it. If you'r not sure what the option is called, refer to the [virtualization options here](https://support.microsoft.com/en-us/windows/enable-virtualization-on-windows-11-pcs-c5578302-6e43-4b4b-a449-8ced115f58e1) (common names are `Virtualization Technology`, `VT-x`, `VT-d`, `AMD-V`, or `SVM`)
    5. Find and select the option to save changes and exit the BIOS / UEFI

>     6. Open Docker Desktop again as instructed in **Option A or B**

Reviewer: ```suggestion
    6. Open Docker Desktop again (as instructed in Option A or Option B) to verify that the error has been resolved
```

    4. Open the start menu and search for "Ubuntu". Start it - it should ask you to create a user with a password. This will be your user to log in to your Ubuntu Linux Subsystem - note down the username and password somewhere secure to make sure you do not forget it.
    5. Open the start menu and search for "Docker Desktop". Start it and go to the Settings. Under the General tab, you will find an option called "Use WSL 2 based engine". Make sure this is checked.

    If opening Docker Desktop from **Option A or B** shows an error, you may need to enable virtualization on your machine

>     The error might look like this:<br><br>

Reviewer: let's be a more specific here:

- the error message should be written out in text in the first paragraph
- "you may need to" -> "you will need to"
- probably we don't need the 2nd paragraph here

Reviewer 2: https://github.com/upleveled/system-setup/pull/16/commits/643325280ecdf96d3411c6c79c6b08871497fb1e
https://github.com/upleveled/system-setup/pull/16/commits/e0a7242d6d3ef4f70c41f5101d9b7a057481974e
https://github.com/upleveled/system-setup/pull/16/commits/0db74a60c1f8c953e516bf024c0f2c568e7bf199

>     <img src="docker-desktop-opening-error.avif"><br><br>

Reviewer: this filename should have `windows-<number>-` as a prefix

Reviewer 2: >     If opening Docker Desktop from **Option A or B** shows an error message that says `Docker Desktop - Windows Hypervisor is not present`, this indicates that virtualization needs to be enabled on your machine

Do I still need the file when I wrote the error message as text?

https://github.com/upleveled/system-setup/pull/16/commits/e0a7242d6d3ef4f70c41f5101d9b7a057481974e

>     Close PowerShell and open it again as administrator (like in step 2)

Reviewer 2: https://github.com/upleveled/system-setup/pull/16/commits/0db74a60c1f8c953e516bf024c0f2c568e7bf199

>     Close PowerShell and open it again as administrator (like in step 2)

Reviewer: what I mean is, it should be step number 1, where the current number 1 is `Restart your computer` (this should become number 2)

>     Close PowerShell and open it again as administrator (like in step 2)

Reviewer 2: b892a20224774ac63333df78e3df02633c1ea849

>     Close PowerShell and open it again as administrator (like in step 2)

Reviewer: > Hmmm, it seems wrong because when they restart their computer, they will lose the PowerShell instance.

the point is not about grouping the steps because it's during a single power cycle - it's about grouping them so that students know where the steps start and end, to make it easier to skip the whole thing, and clear that they are now in the instructions for trying to deal with the Docker error

>     Close PowerShell and open it again as administrator (like in step 2)

Reviewer 2: bfb2c747b5f11179451e3f22e6c9344e007956d0

>     Check if your computer supports virtualization. Run this command:

Reviewer: this should follow the style of the other instructions to enter commands in the guide

Reviewer 2: > is it possible to limit the number of properties being displayed? are all 5 really relevant?

All five are not necessarily relevant, but I just wanted to keep the expected output the way it is so it doesn't feel confused

https://github.com/upleveled/system-setup/pull/16/commits/493e54b7da21449f98e031dc87aead474b1a121d
https://github.com/upleveled/system-setup/pull/16/commits/63fe30ab3bddc4701ef0e54ac5d1981c9ee84d24

Reviewer: > All five are not necessarily relevant, but I just wanted to keep the expected output the way it is so it doesn't feel confused

I agree that we should display the output exactly as it appears on the screen for students, but I think we can probably limit the number of lines of output to make it easier for students to check each line

eg. [piping to `Select-Object` with commas: `Get-ComputerInfo -Property * | Select-Object CsName,CsModel`](https://community.spiceworks.com/t/get-computerinfo-cmdlet-all-servers/712473)

Reviewer 2: The command has been improved

https://github.com/upleveled/system-setup/commit/2465d51de575c0d7a4b0bbd1b307117bfab332cb

    ```bash
    Get-ComputerInfo -Property HyperV*
    ```

    This should display the following output:

    ```bash
    HyperVisorPresent: False
    HyperVRequirementDataExecutionPreventionAvailable : True
    HyperVRequirementSecondLevelAddressTranslation    : True
    HyperVRequirementVirtualizationFirmwareEnabled    : False
    HyperVRequirementVMMonitorModeExtensions          : True
    ```

    Check if virtualization is enabled. Run this command:

    ```bash
    systeminfo | findstr Virtualization
    ```

    This should display the following output:

    ```bash
    Virtualization Enabled In Firmware: No

>     ```

Reviewer: can this be combined with the previous command?

Reviewer 2: I think this can be removed as the information it provides is also available on the first command output
https://github.com/upleveled/system-setup/pull/16/commits/63fe30ab3bddc4701ef0e54ac5d1981c9ee84d24

>     Continue with the steps below if virtualization is not enabled to enable virtualization in your BIOS/UEFI:

Reviewer: - I don't think that it needs to say "continue" here, it's natural to keep following the steps if nothing goes wrong
- there should be some text to describe what to do if either of the "outputs" above are different (and what that means)
- maybe we still want to separate these steps to enable virtualization in BIOS / UEFI

Reviewer 2: > maybe we still want to separate these steps to enable virtualization in BIOS / UEFI (but maybe it's not important to have them separated from step one "Close PowerShell ..." above)

I don't think we should make it a separate point as it is not something that all students would need to do. I think I did had this issue because I moved from Windows 10 to 11. According to [this docs](https://support.microsoft.com/en-us/windows/enable-virtualization-on-windows-11-pcs-c5578302-6e43-4b4b-a449-8ced115f58e1), it seems most computers should have it already enabled.

The Continue with the... statement is combined with the new one on the top as I think they should continue with anything on these steps only if they have an error

https://github.com/upleveled/system-setup/pull/16/commits/493e54b7da21449f98e031dc87aead474b1a121d
https://github.com/upleveled/system-setup/pull/16/commits/63fe30ab3bddc4701ef0e54ac5d1981c9ee84d24

>     1. Restart your computer

Reviewer: "computer" -> "machine", to match style of the guide

Reviewer 2: a2135c94755b953e270dbc19b81d5dae9a252cc9

>     2. Before the brand logo appears, press the BIOS/UEFI access key repeatedly. Common options are <kbd>Delete</kbd>, <kbd>Esc</kbd>, <kbd>F1</kbd>, <kbd>F2</kbd>, <kbd>F4</kbd>, <kbd>F10</kbd>, or <kbd>F12</kbd> or as shown on the startup screen or check this [BIOS/UEFI key lists](https://www.disk-image.com/faq-bootmenu.htm)

Reviewer: ```suggestion
    2. As soon as the monitor turns black during restart, press the BIOS key or UEFI key for your machine repeatedly. If you're not sure what that key is, either try to read the key on the screen quickly as your machine restarts or refer to [hotkey documentation online](https://www.disk-image.com/faq-bootmenu.htm) (common keys are <kbd>Delete</kbd>, <kbd>Esc</kbd>, <kbd>F1</kbd>, <kbd>F2</kbd>, <kbd>F9</kbd>, <kbd>F10</kbd> or <kbd>F12</kbd>)
```

>     2. Before the brand logo appears, press the BIOS/UEFI access key repeatedly. Common options are <kbd>Delete</kbd>, <kbd>Esc</kbd>, <kbd>F1</kbd>, <kbd>F2</kbd>, <kbd>F4</kbd>, <kbd>F10</kbd>, or <kbd>F12</kbd> or as shown on the startup screen or check this [BIOS/UEFI key lists](https://www.disk-image.com/faq-bootmenu.htm)

Reviewer: I think that this `disk-image.com` link doesn't have many options, it's missing big common PC manufacturers like MSI

check into the largest popular PC manufacturers and find a link that is better

you can adjust my Google search query here if you find any other manufacturers that are important:

```
hot keys for bios "msi" samsung lenovo hp dell acer asus sony -site:youtube.com
```

Reviewer 2: This is most comprehensive
1ae71beaf0ac80a5ab349ad8ea834a356133e4b1

>     3. Navigate to the `BIOS/UEFI Setup Utility`

Reviewer: what does this mean?

Reviewer 2: In some computers, like the ones I am testing on, I will need to press the `F10` key to navigate to the BIOS Setup Utility screen, where I can enable the virtualization option

In Lukas's computer, he goes straight into that without needing to press any key. When you need to press a key, you will see something like this 

`F10 BIOS Setup` or `BIOS Setup F10`

Reviewer: ah ok, let's try to be clear that this is an optional step - that students *may* need to find and select an option to enter the BIOS / UEFI

>     4. Find the virtualization option for your computer and enable it

Reviewer: let's:

- be a bit more clear that the student will need to search for it
- check if there are any "more common" names that we can add here, just like we added "more common" hotkeys above
- offer a link to some good documentation about what the names of this would be

Reviewer 2: I can't find any documentation that centralizes all the virtualization options. I have the link from Microsoft, but they will still need to click around on the lists of the computer manufacturers there.
https://support.microsoft.com/en-us/windows/enable-virtualization-on-windows-11-pcs-c5578302-6e43-4b4b-a449-8ced115f58e1

Some other options were
https://techlibrary.hpe.com/docs/iss/proliant_uefi/UEFI_TM_030617/s_Virtualization_options.html

f467126fa9850475d169fe0cdc53aa0d9ca4c7e7

>     5. Save changes and exit the BIOS/UEFI

Reviewer: ```suggestion
    5. Find the option to save changes and exit the BIOS / UEFI
```

> 

Reviewer: it seems there's a step missing here to retry the previous thing that failed

Reviewer 2: 8fb18541d31b610f9133a20abe78b596374469e2

    ```

    It should print out a welcome message like this:<br><br>
    <img src="macos-7-docker-hello-world.png"><br><br>

>     If opening Docker Desktop or running the command above shows a message that says `Docker Desktop - Windows HyperVisor is not Present`, you may need to enable virtualization on your machine.

Reviewer: 1. let's make sure that we start this as early as the first error could occur (check if there is any step before this where it would be better to start this guide)
2. check if we should break this up into a separate optional step, to make it easier for people to skip this section

Reviewer 2: I moved it after the Docker Desktop installation because the error would happen when they try to open Docker Desktop after the installation

https://github.com/upleveled/system-setup/pull/16/commits/d98cc2a398beaeea06481e1f957032a6bbd27d81

I was thinking that they can already skip after reading this first line

> If opening Docker Desktop from **Option A or B** shows an error, you may need to enable virtualization on your machine

    The error might look like this:<br><br>
    <img src="docker-desktop-opening-error.avif"><br><br>

    First check if virtualization is enabled on your computer: To do this, open PowerShell as an administrator and run the following command:
    ```bash

>     systeminfo | findstr /C:"Hyper-V" /C:"Virtualization"

Reviewer: - do we need the extra `Hyper-V` check here?
- do we need the `/C:`? can we just do `systeminfo | findstr Virtualization`?

Reviewer 2: We don't need the `/C:`
I did the command like this `systeminfo | findstr "Hyper-V Virtualization"` instead so that we can search for the two instances.  I expect that different Windows computers will call this functionality either `Hyper-V` or `Virtualization`, and I would want to search for both. However, 

in my Windows 10 and 11 Pro with Lukas's 10 Home, It is all called `Hyper-V` as searching for `Virtualization` did not produce any output

Maybe we should just have this `systeminfo | findstr Hyper-V`

62664824c1626b2307eb9fa836b38f582bcdb57a

    ```

    This should display the output below: Note the `Virtualization Enabled In Firmware: No`, this shows that virtualization is not enabled on your computer

    ```bash
    Hyper-V Requirements: VM Monitor Mode Extensions: Yes
                          Virtualization Enabled In Firmware: No
    ```

    Continue with the steps below if virtualization is not enabled, Enable virtualization in your BIOS or UEFI:

>     1. Start your computer if it is off, or restart it if it has already started

Reviewer: spend a few edit cycles to try to reduce the length of the whole text

one example of possible simplification of this line: instead of saying "either do this or that", and building a concept of state into it, just give one instruction that is probably going to be the case for 90% of students (in this case "Restart your computer")

Reviewer 2: Overall instructions changed

e9dcf99d92e43034c3071d66ca2dd15ccfe271b0

>     2. Right when the computer is starting up (still on the black screen before the computer brand logo flashes), press repeatedly any of these keys: <kbd>Delete</kbd>, <kbd>Esc</kbd>, <kbd>F1</kbd>, <kbd>F2</kbd>, <kbd>F4</kbd>, <kbd>F10</kbd>, or <kbd>F12</kbd>. Each computer manufacturer uses a different key but it may show a brief message at boot time telling you which one to press. If you miss it the first time, restart and try again. If you are not able to enter the BIOS via this method, check this [BIOS key lists](https://www.disk-image.com/faq-bootmenu.htm) for different computers or consult your computers manual

Reviewer 2: Steps refactored

https://github.com/upleveled/system-setup/pull/16/commits/e9dcf99d92e43034c3071d66ca2dd15ccfe271b0

    3. If you are not already in the `BIOS Setup Utility` interface, you will need to press the key next to any option that says `BIOS Setup` or has `Bios`, `UEFI` in it to open BIOS Setup Utility
    4. Find the virtualization option in the BIOS Setup Utility. This can be under the `Configuration`, `Advanced` or `Security` tab, you will likely navigate with the `Down`, `Up`, `Left`, or `Right` arrow keys in this mode if the mouse doesn't work

>     5. Enable virtualization by setting `Disabled` to `Enabled` on the Virtualization option. The Virtualization option can be called: `Virtualization Technology`, `SVM(Secure Virtual Machine) Mode`, `AMD-V(AMD Virtualization)`, `AMD SVM`, `Intel VTX(Intel Virtualization Technology)`, `Intel VT-d(Virtualization for Direct-IO)` or `AMD IOMMU(Input/Output Memory Management Unit)` depending on your computer manufacturer

Reviewer 3: Do we need to write `Disabled` to `Enabled`? can we write "Enable virtualization by setting it to `Enabled` on the Virtualization option"?

Reviewer 2: Steps refactored

https://github.com/upleveled/system-setup/pull/16/commits/e9dcf99d92e43034c3071d66ca2dd15ccfe271b0

    6. Find option that says `Save Changes and Exit` or anything that signifies that the changes you made will be saved and press `Enter` or the key next to that option. Your computer will restart

    If you are unable to find the virtualization settings in your BIOS, ensure that your computer supports virtualization. To do this, open PowerShell as an administrator and run the following command:

    ```bash
    Get-ComputerInfo -Property 'HyperV*'

>     ```

Reviewer: shouldn't this check be done much earlier? (to avoid students spending a lot of effort only to find out their computer doesn't have the option)

Reviewer 2: https://github.com/upleveled/system-setup/pull/16/commits/5ddf00a704e52ec95785d9f1f0eac75a32f79e20

    ```

    It should print out a welcome message like this:<br><br>
    <img src="macos-7-docker-hello-world.png"><br><br>

>     if opening Docker Desktop or running the command above shows a message that says `Docker Desktop - Windows HyperVisor is not Present`, you may need to enable virtualization on your machine.

Reviewer 3: `If` uppercase?

Reviewer 2: 9117d0098ffebe918a3bb4c0ce0df6064f61fa37

    The error might look like this:<br><br>
    <img src="docker-desktop-opening-error.avif"><br><br>

    First check if virtualization is enabled on your computer on the **Performance** tab of the Task Manager (open the Task Manager with <kbd>Ctrl</kbd>-<kbd>Shift</kbd>-<kbd>Escape</kbd>):<br><br>

    <img src="windows-7-virtualisation-enabled.avif"><br><br>

    Continue with the steps below if virtualization is not enabled, Enable virtualization in your BIOS or UEFI:

>     1. Restart your computer if it is already started or start it, if not

Reviewer 3: the `,if not` seems weird to me when I read it, maybe something like 
"Start your computer or restart it if it's already running"?

Reviewer 2: 1c05c6beaef89c67dab38aee848d86f849357db8

>     2. Right when the computer is starting up (Still on the black screen befor the computer brand logo blashes), press repeatedly any of these keys: <kbd>Delete</kbd>, <kbd>Esc</kbd>, <kbd>F1</kbd>, <kbd>F2</kbd>, <kbd>F4</kbd>, <kbd>F10</kbd>, or <kbd>F12</kbd>. Each computer manufacturer uses a different key but it may show a brief message at boot time telling you which one to press. If you miss it the first time, restart and try again. If you are not able to enter the BIOS via this method, check this [BIOS key lists](https://www.disk-image.com/faq-bootmenu.htm) for different computers or consult your computers manual

Reviewer 3: `Still` lowercase, and `before` typo

Reviewer 2: > For me, when I see the brand logo of my motherboard, I can still press the keys to open BIOS

I need to start pressing the BIOS button in the two computers I used before the logo shows; if not, the computer will have a normal boot. If you can start the BIOS mode by pressing the key before the logo shows, I will leave it like that, as that would be when to start initializing BIOS in most computers

`blashes` meant `flashes`
ab0172064419fd1490faa93d3f593f58dfe08d41

48b4e634c0800e7304f89cc2d21482266540de1b

Reviewer 3: I created a small video and added it to my [comment](https://github.com/upleveled/system-setup/pull/16#pullrequestreview-1988781209)

Reviewer 2: Yes, thank you so much. Your computer shows brief instructions on what key to press when booting. I think this is covered in the statement I made here

> Each computer manufacturer uses a different key, but it may show a brief message at boot time telling you which one to press

>     3. Press the key next to any option that says `BIOS Setup` or has `Bios`, `UEFI` in it to open BIOS Setup Utility

Reviewer 3: I don't have to press any key to open BIOS Setup after pressing `F2` or `Delete` to open BIOS. I have to go to the "Advanced Mode" and this is explained in the next step

Reviewer 2: When your machine is booting, and you press the `F2` or `Delete` button, Is it possible for you to share the screen it takes you to here?

For some computers, it is a screen like this, which is what I also have on my computer and on Anna's computer, which is `Windows 10 Pro`.  on both computers, I needed to press the `F10` key to continue into the BIOS

![startup menu](https://github.com/upleveled/system-setup/assets/74430629/4f8fbe4c-35c8-4eb5-acd6-13e319384df9)

Reviewer 3: I don't know how to share the screen, maybe I can take a video with my phone? 
> For some computers, it is a screen like this, which is what I also have on my computer and on Anna's computer, which is Windows 10 Pro. on both computers, I needed to press the F10 key to continue into the BIOS

This is not the case with my machine. I guess it depends on the Hardware and the BIOS from the different companies. I see the ASUS Logo and press `F2` or `DELETE` and I am instantly in the BIOS.

Reviewer 2: Yes, you can record or take a picture with your phone. I have updated the point to accommodate the concern you raised

dfb059721d3bc87b8bfaff215fbe8248ff90b758

>     4. Find the virtualization option in the BIOS Setup Utility. This can be under the `Configuration`, `Advanced` or `Security` tab, you can only navigate with the `Down`, `Up`, `Left`, or `Right` keys in this mode

Reviewer 3: I can also use the mouse in my BIOS

    ```
    This step will change the default Git branch from `master` to `main` (see https://github.com/github/renaming).<br><br>
29. Copy and run each of these lines separately in Hyper.<br><br>
    ```bash
    git config --global core.autocrlf false
    git config --global core.eol lf
    ```
    This step will improve line breaks compatibility on Windows.
30. Go back to GitHub, and go to your profile page by clicking on your avatar at the top right and selecting **Your profile**<br><br>
    <img src="./general-github-your-profile.png"><br><br>
    Copy the `github.com/...` URL in the address bar of your browser, for use in the next step.
31. Open the Start menu and start Slack. Log in to the UpLeveled Slack. Send your GitHub profile URL to [Lukas](https://upleveled.slack.com/team/U026UPA0TKL). Also send your Google Account email address to Lukas (if you haven't already).
32. <a name="specs"></a>Open the start menu, type "Settings" and open the app (or click on the cog on the left). Select "System" and "About". Under "Device specifications", click the Copy button and paste this to Lukas. Under "Windows specifications", click the Copy button and paste this to Lukas.

> 33. On your phone, go to the app store and install Slack on your phone. Log in to the UpLeveled Slack.

Reviewer: seems like these changes are unrelated and should be reverted?

> 33. On your phone, go to the app store and install Slack on your phone. Log in to the UpLeveled Slack.

Reviewer: seems like these changes are unrelated?

if the changes here are about the numbering, we can pull this out into a separate PR and merge it quickly

Reviewer 2: New PR
https://github.com/upleveled/system-setup/pull/66
